src/util/util.rs

```diff
@@ -40,6 +40,26 @@ pub fn get_uinput_device_name(dev_file_path: &String) -> Result<Option<String>>
 	}
 }
 
+fn insert_device_name_path_pair(map: &mut HashMap<String, String>, name: String, path: String) {
+	if !map.contains_key(&name) {
+		// Map doesn't contain a device by this name -- just insert it and return
+		map.insert(name, path);
+		return;
+	}
+
+	// Map contains a device by this name, append an incrementing number until there is no collision
+	let mut i = 1;
+	loop {
+		let fixed_name = format!("{} ({})", name, i);
+		if !map.contains_key(&fixed_name) {
+			map.insert(fixed_name, path);
+			break;
+		} else {
+			i += 1
+		}
+	}
+}
+
 pub fn get_all_uinput_device_names_to_paths() -> Result<HashMap<String, String>> {
 	let paths = fs::read_dir("/dev/input/")?;
```

Comment:
Maybe the `paths` should be sorted in order to ensure that the numbering that is added is deterministic?

Alternatively, I did something simliar [here](https://github.com/samvel1024/kbct/pull/50) by just appending the device number instead